Component to add appointments.

JaredDsD opened the pull request:
Work in progress, I still need to make some corrections on the code, I get some errors.

Please review my code and provide feedback.

JaredDsD committed:
Added dependancies.

JaredDsD committed:
Added dependancies.

JaredDsD committed:
Resolving the conflics

JaredDsD committed:
Updating appointments form

JaredDsD committed:
Updating appointments form

JaredDsD committed:
Moved appointmentForm to be able to display it

JaredDsD committed:
Moved Schema to test the visual appearence of the component and added dependencies.

JaredDsD committed:
Moved Schema to test the visual appearence of the component and added dependencies.

Ralexs0096 commented on package.json:
```diff
@@ -14,16 +14,21 @@
   },
   "dependencies": {
+    "@date-io/date-fns": "^3.2.1",
     "@emotion/react": "^11.14.0",
     "@emotion/styled": "^11.14.0",
+    "@hookform/resolvers": "^5.0.1",
     "@mui/icons-material": "^7.1.0",
-    "@mui/material": "^7.1.0",
+    "@mui/material": "^7.1.1",
     "@mui/styled-engine-sc": "^7.1.0",
+    "@mui/x-date-pickers": "^8.5.1",
     "@vitejs/plugin-react": "^4.5.0",
+    "dayjs": "^1.11.13",
     "formik": "^2.4.6",
     "next": "15.3.2",
     "react": "^19.0.0",
     "react-dom": "^19.0.0",
+    "react-hook-form": "^7.56.4",
```
we have a problem here @JaredDsD, We are adding more libraries but this was not agreed upon by the entire team.

Why is this important? Because the libraries you add have to be understood and in many cases, learned by the whole team.

Additionally, you are adding `react-hook-form` when we already have `formik`.

cc: @AlvisonHunterArnuero

Ralexs0096 replied:
`dayjs` and `date-fns` are both for date management, if the intention is to add a `time provider`, you have to choose 1, I would suggest `date-fns` over `dayjs`.

AlvisonHunterArnuero replied:
We mentioned this before, team: We cannot simply add dependencies without consulting the rest of the team about this. let's keep this in mind for future endeavors, please.

Ralexs0096 commented on src/app/add-appointments/page.tsx:
```diff
@@ -0,0 +1,175 @@
+
+import { useRouter } from "next/navigation";
+import { Box, Grid, TextField, Button } from "@mui/material";
+//import { DatePicker, TimePicker } from "@mui/x-date-pickers";
+//import { Controller, useForm } from "react-hook-form";
+//import { yupResolver } from "@hookform/resolvers/yup";
+import { appointmentSchema } from "./AppointmentSchema";
+//import dayjs, { Dayjs } from "dayjs";
```
please remove the comments. this is similar to leave one of this:.

![image](https://github.com/user-attachments/assets/fa3f8d68-d183-41bb-a719-c125ce5c76b0)

Ralexs0096 commented on src/app/add-appointments/page.tsx:
```diff
@@ -0,0 +1,175 @@
+const AppointmentForm = () => {
+  const router = useRouter();
+
+  const {
+    control,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<FormData>({
+    resolver: yupResolver(appointmentSchema),
+    defaultValues: {
+      title: "",
+      date: dayjs(),
+      startTime: dayjs(),
+      endTime: dayjs(),
+      description: "",
+    },
+  });
```
I don't understand why we added RHF when we already had formik. 🌵

AlvisonHunterArnuero replied:
Great question: Why was this added to the project @JaredDsD ?

Ralexs0096 commented on src/app/add-appointments/page.tsx:
```diff
@@ -0,0 +1,175 @@
+    };
+
+    try {
+      const res = await fetch(
+        `${process.env.NEXT_PUBLIC_API_URL}/appointments`,
+        {
+          method: "POST",
+          headers: { "Content-Type": "application/json" },
+          body: JSON.stringify(payload),
+        }
+      );
+
+      if (!res.ok) throw new Error("Error creating appointment");
+
+      alert("Appointment created!");
+      router.push("/appointments");
```
can we make an `action` for this? 🤔

Ralexs0096 commented:
It would be great if you could add some images

JaredDsD committed:
I'm refactoring my component, I'm still working.

JaredDsD committed:
I'm refactoring my component, I'm still working.

JaredDsD committed:
Refactor AppointmentForm and AppointmentSchema for improved readability and functionality

JaredDsD committed:
Merge branch 'develop' into jared_new_appointment

JaredDsD committed:
feat: integrate jsonwebtoken for user authentication and enhance appointment management

- Added jsonwebtoken and its types to package.json for handling JWTs.
- Refactored appointment creation to use a server action for better API interaction.
- Implemented user fetching and email extraction from JWT for authenticated user context in appointment forms.
- Updated appointment services to handle JWT in requests and improved error handling.
- Removed deprecated models and services, consolidating appointment-related logic.
- Enhanced navigation and user experience with a logout feature and redirection based on authentication state.

JaredDsD commented:
Hello Tribe, I was finally able to make this thing work. 
There are many things that I did, I had to refactorize almost all, adapt and change some of the files created by the other guys. 

![Captura de pantalla 2025-06-21 234023](https://github.com/user-attachments/assets/b5505e60-a2a7-4284-b2fe-1038736def28)
![Captura de pantalla 2025-06-22 002230](https://github.com/user-attachments/assets/16148466-55b4-4f8b-9ac4-289677ac58d8)
![Captura de pantalla 2025-06-22 002255](https://github.com/user-attachments/assets/0de29496-5ca5-4623-8f0e-ab00a1cd77c0)

You can see the funtionality in the pictures, I can still improve it of course, I'll apreciate all the suggestions.

JaredDsD committed:
Removing comments

Cema2019 commented:
The updates have already been manually implemented on the branch develop-test.
Proceeding to close this PR as requested.
